### PR TITLE
New command: `[register] UNLIMITED POWER [value/register]`

### LINKED ISSUE #47: [Command suggestion] `[register] unlimited power [value/register]`
**Command Syntax:** (e.g. `stonks [register]`) 
x unlimited power y
x^y

**Command Description:**
basic power operation on numbers.

Which x86_64 Assembly instruction(s) would be used to translate this command?
the words mason, what do they mean?

What could this command be useful for?
power

What Meme is your command based on? (You can also provide a link to e.g. its KnowYourMeme page)
unlimited power
https://knowyourmeme.com/memes/darth-sidious-unlimited-power

### PR DESCRIPTION
**Pull request by kammt**
Closes #47

### CODE REVIEW

**Comment by xarantolus** [comment]
I think there's a quirk in how this command is handling ascii characters as inputs.

Take for example this program:

```
I like to have fun, fun, fun, fun, fun, fun, fun, fun, fun, fun main
    rax is brilliant, but I like 2
    rax UNLIMITED POWER 2
    
    what can I say except al

    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax

    what can I say except al


    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax

    what can I say except al


    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax

    what can I say except al


    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax

    what can I say except al


    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax

    what can I say except al


    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax

    what can I say except al


    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax
    they had us in the first half, not gonna lie rax

    what can I say except al
    
    I see this as an absolute win
```

It should calculate 2^2 and output the bytes in reverse order. Run it with the following command: `memeasm -o './main' 'test.memeasm' && ./main | hexdump -C`.

Since we output the bytes in reverse order, we would expect an output like this:

```
00000000  04 00 00 00 00 00 00 00                           |........|
00000008
```

However, the actual output is like this:
```
00000000  00 00 00 00 00 00 04 00                           |........|
00000008
```

The program seems to have calculated 2^50, not 2^2. In fact, when we change the `rax UNLIMITED POWER 2` to  `rax UNLIMITED POWER 50`, we get the *exact* same output as with `2`.

The reason for this is that the single character `2` is interpreted as ascii `2`, which means that the value used as the power is its value in the ascii table: `50`.

When changing the command to `rax UNLIMITED POWER 02` the compiler interprets it as decimal value `2`, which one would expect.

Now that in itself is not necessarily a bug, it just has the same behavior as e.g. the `what can I say except` command where `what can I say except 2` should output the character `2`. It is however inconsistent with a command like `rax is brilliant, but I like 2`, where the `2` is considered as the numerical value.


TL;DR: Single digit numbers are interpreted as their ascii values (so `2` is actually the value `50`)

**Comment by kammt** [comment]
You're right! Wow you found that issue fast :D
TIL: I really need to simplify the parameter types system, using binary constants is confusing
